core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java

```diff
@@ -124,4 +160,25 @@ public int maxFailoverAttempts() {
          return maxFailoverCount;
       }
    }
+
+   static class SameNodeTaskFailoverPolicy implements DistributedTaskFailoverPolicy {
+      private final int maxFailoverCount;
```

Comment:
Also unfortunately ++ is not atomic (which means you could lose an upate)  You would need to use an AtomicInteger instead which allows for `incrementAndGet`